openssl-dynamic/src/main/c/ssl.c

```diff
@@ -1777,6 +1777,142 @@ TCN_IMPLEMENT_CALL(void, SSL, freeX509Chain)(TCN_STDARGS, jlong x509Chain)
     sk_X509_pop_free(chain, X509_free);
 }
 
+TCN_IMPLEMENT_CALL(void, SSL, setKeyMaterialServerSide)(TCN_STDARGS, jlong ssl, jlong chain, jlong key)
+{
+
+#if OPENSSL_VERSION_NUMBER < 0x10002000L || defined(LIBRESSL_VERSION_NUMBER)
+    // Only supported on openssl 1.0.2+
+    tcn_Throw(e, "Not supported");
+#else
+    SSL *ssl_ = J2P(ssl, SSL *);
+
+    TCN_CHECK_NULL(ssl_, ssl, /* void */);
+
+    EVP_PKEY* pkey = J2P(key, EVP_PKEY *);
+    STACK_OF(X509) *cchain = J2P(chain, STACK_OF(X509) *);
+    int numCerts;
+    X509* x509 = NULL;
+    char err[ERR_LEN];
+    int n;
```

Comment:
nit: we use `i` for an increment variable in the other method ... consider being consistent.